src/Utils/Mappers.php

```diff
@@ -0,0 +1,195 @@
+            $multivariateFeatureStateValues = $featureState->multivariate_feature_state_values ?? [];
+            uksort($multivariateFeatureStateValues, fn ($a, $b) => $a->id <=> $b->id);
+            foreach ($multivariateFeatureStateValues as $multivariateFeatureStateValue) {
+                $variant = new FeatureValue();
+                $variant->value = $multivariateFeatureStateValue->multivariate_feature_option->value;
+                $variant->weight = $multivariateFeatureStateValue->weight;
+                $feature->variants[] = $variant;
+            }
+
+            $featureContexts[$feature->name] = $feature;
+        }
+        
+        return $featureContexts;
+    }
+
+    /**
+     * @param array<object> $identityOverrides
+     * @return array<string, SegmentContext>
+     */
+    private static function _mapIdentityOverridesToSegments($identityOverrides)
+    {
+        /** @var array<string, array<string>> */
+        $featuresToIdentifiers = [];
+        foreach ($identityOverrides as $identityOverride) {
+            $identityFeatures = $identityOverride->identity_features ?? [];
+            uksort($identityFeatures, fn ($a, $b) => strcasecmp($a->feature->name, $b->feature->name));
+            if (empty($identityFeatures)) {
+                continue;
+            }
+
+            /** @var array<array<mixed>> */
+            $overridesKey = [];
+            foreach ($identityFeatures as $featureState) {
+                $part = [
+                    $featureState->feature->id,
+                    $featureState->feature->name,
+                    $featureState->enabled,
+                    $featureState->feature_state_value,
+                ];
+                $overridesKey[] = $part;
+            }
+            $featuresToIdentifiers[serialize($overridesKey)][] = $identityOverride->identifier;
+        }
+
+        /** @var array<string, SegmentContext> */
+        $segments = [];
+        foreach ($featuresToIdentifiers as $serializedoverridesKey => $identifiers) {
+            $segment = new SegmentContext();
+            $segment->key = hash('sha256', $overridesKey); 
+            $segment->name = "identity_overrides";
+
+            $identifiersCondition = new SegmentCondition();
+            $identifiersCondition->property = "$.identity.identifier";
+            $identifiersCondition->operator = SegmentConditionOperator::IN;
+            $identifiersCondition->value = $identifiers;
+
+            $identifiersRule = new SegmentRule();
+            $identifiersRule->type = SegmentRuleType::ALL;
+            $identifiersRule->conditions = [$identifiersCondition];
+            $segment->rules = [$identifiersRule];
+
+            $segment->overrides = [];
+            foreach (unserialize($overridesKey) as $overideKey) {
+                [$featureKey, $featureName, $enabled, $value] = $overideKey;
+                $feature = new FeatureContext();
+                $feature->key = "--irrelevant--";
```

Comment:
I'd rather have something meaningful here, though I need to seek education. This is to force my own revisit, though it is welcome to discussion.